include/streaming_protocol/Types.h

```diff
@@ -137,6 +137,13 @@ class Resolution {
     std::string toString() const;
 };
 
+/// signals following an implicit rule produce indexed values that consist of the signal value and a value index
+template<class Type>
+struct IndexedValue
+{
+    uint64_t index;
+    Type value;
+};
```

Comment:
To safely use `sizeof(IndexedValue<>)` here you would need `__attribute__((packed))` on the struct. Otherwise the compiler can insert padding depending on `Type` and on the platform.